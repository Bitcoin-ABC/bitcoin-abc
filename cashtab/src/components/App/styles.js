// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled, { createGlobalStyle, css } from 'styled-components';
import { ToastContainer } from 'react-toastify';

export const ExtensionFrame = createGlobalStyle`
    html, body {
        min-width: 400px;
        min-height: 600px;
    }
`;

export const GlobalStyle = createGlobalStyle`
    *::placeholder {
        color: ${props => props.theme.forms.placeholder} !important;
    }
    a {
        color: ${props => props.theme.eCashBlue}
        &:hover {
            color: ${props => props.theme.eCashPurple}
            text-decoration: none;
        }
    }
`;

export const CashtabNotification = styled(ToastContainer)`
    .Toastify__progress-bar-theme--dark {
        background: #00abe7;
    }
    .Toastify__progress-bar-theme--light {
        background: #00abe7;
    }
`;

export const CustomApp = styled.div`
    text-align: center;
    font-family: 'Poppins', sans-serif;
    background-color: ${props => props.theme.backgroundColor};
    background-size: 100px 171px;
    background-image: ${props => props.theme.backgroundImage};
    background-attachment: fixed;
    min-height: 100vh;
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;
export const WalletBody = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
`;
const AppPositionCss = css`
    width: 500px;
    background: ${props => props.theme.walletBackground};
    -webkit-box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    -moz-box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    box-shadow: 0px 0px 24px 1px ${props => props.theme.shadow};
    @media (max-width: 768px) {
        width: 100%;
        -webkit-box-shadow: none;
        -moz-box-shadow: none;
        box-shadow: none;
    }
`;

export const WalletCtn = styled.div`
    ${AppPositionCss}
    position: relative;
    padding: 0 0 100px;
    min-height: calc(100vh - 80px);
`;

export const Footer = styled.div`
    ${AppPositionCss}
    z-index: 2;
    height: 80px;
    border-top: 1px solid rgba(255, 255, 255, 0.5);
    position: fixed;
    bottom: 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0;
`;

export const NavWrapper = styled.div`
    width: 100%;
    height: 100%;
    cursor: pointer;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 1.3rem;
    margin-bottom: 5px;
`;

export const NavIcon = styled.span`
    @media (hover: hover) {
        ${NavWrapper}:hover & {
            background-color: ${props =>
                props.clicked ? 'transparent' : props.theme.eCashPurple};
            ::before,
            ::after {
                background-color: ${props => props.theme.eCashPurple};
            }
        }
    }

    position: relative;
    background-color: ${props =>
        props.clicked ? 'transparent' : props.theme.buttons.primary.color};
    width: 2rem;
    height: 2px;
    display: inline-block;
    transition: transform 300ms, top 300ms, background-color 300ms;
    &::before,
    &::after {
        content: '';
        background-color: ${props => props.theme.buttons.primary.color};
        width: 2rem;
        height: 2px;
        display: inline-block;
        position: absolute;
        left: 0;
        transition: transform 300ms, top 300ms, background-color 300ms;
    }
    &::before {
        top: ${props => (props.clicked ? '0' : '-0.8rem')};
        transform: ${props => (props.clicked ? 'rotate(135deg)' : 'rotate(0)')};
    }
    &::after {
        top: ${props => (props.clicked ? '0' : '0.8rem')};
        transform: ${props =>
            props.clicked ? 'rotate(-135deg)' : 'rotate(0)'};
    }
`;

export const NavMenu = styled.div`
    position: fixed;
    margin-right: 125px;
    bottom: 80px;
    display: flex;
    width: 225px;
    flex-direction: column;
    border: ${props => (props.open ? '1px solid' : '0px solid')};
    border-color: ${props =>
        props.open ? props.theme.contrast : 'transparent'};
    justify-content: center;
    align-items: center;
    @media (max-width: 768px) {
        right: 0;
        margin-right: 0;
    }
    overflow: hidden;
    transition: ${props =>
        props.open
            ? 'max-height 250ms ease-in-out , border-color 250ms ease-in-out, border-width 250ms ease-in-out'
            : 'max-height 250ms cubic-bezier(0, 1, 0, 1), border-color 250ms ease-in-out, border-width 250ms ease-in-out'};
    max-height: ${props => (props.open ? '100vh' : '0')};
`;

export const NavItem = styled.button`
    display: flex;
    justify-content: space-between;
    text-align: left;
    font-size: 24px;
    padding: 12px;
    align-items: center;
    width: 100%;
    white-space: nowrap;
    background-color: ${props => props.theme.walletBackground};
    border: 1px solid ${props => props.theme.walletBackground};
    color: ${props => props.theme.contrast};
    gap: 6px;
    cursor: pointer;
    &:hover {
        color: ${props => props.theme.eCashPurple};
        svg,
        g,
        path {
            fill: ${props => props.theme.eCashPurple};
        }
    }
    svg {
        fill: ${props => props.theme.contrast};
        max-width: 33px;
        height: auto;
        flex: 1;
    }
    g,
    path {
        fill: ${props => props.theme.contrast};
    }
    p {
        flex: 2;
        margin: 0;
    }
    ${({ active, ...props }) =>
        active &&
        `    
        color: ${props.theme.navActive};
        svg, g, path {
            fill: ${props.theme.navActive};
        }
  `}
`;

export const NavButton = styled.button`
    :focus,
    :active {
        outline: none;
    }
    @media (hover: hover) {
        :hover {
            svg,
            g,
            path {
                fill: ${props => props.theme.eCashPurple};
            }
        }
    }
    width: 100%;
    height: 100%;
    cursor: pointer;
    padding: 0;
    background: none;
    border: none;
    font-size: 10px;
    svg {
        fill: ${props => props.theme.contrast};
        width: 30px;
        height: 30px;
    }
    g,
    path {
        fill: ${props => props.theme.contrast};
    }
    ${({ active, ...props }) =>
        active &&
        `    
        color: ${props.theme.navActive};
        svg, g, path {
            fill: ${props.theme.navActive};
        }
  `}
`;
export const ScreenWrapper = styled.div`
    padding: 0px 30px;
    @media (max-width: 768px) {
        padding: 0px 15px;
    }
`;

export const HeaderCtn = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 15px 0;
    gap: 6px;
`;

export const CashtabLogo = styled.img`
    width: 120px;
    @media (max-width: 768px) {
        width: 110px;
    }
`;

// Easter egg styled component not used in extension/src/components/App.js
export const EasterEgg = styled.img`
    position: fixed;
    bottom: -195px;
    margin: 0;
    right: 10%;
    transition-property: bottom;
    transition-duration: 1.5s;
    transition-timing-function: ease-out;

    :hover {
        bottom: 0;
    }

    @media screen and (max-width: 1250px) {
        display: none;
    }
`;

export const NavHeader = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 1rem;
    color: ${props => props.theme.navActive};
    svg {
        padding: 0.2rem;
        fill: ${props => props.theme.navActive};
        height: 33px;
        width: 30px;
    }
    g {
        fill: ${props => props.theme.navActive};
    }
    path {
        fill: ${props => props.theme.navActive};
    }
`;

export const BalanceHeaderContainer = styled.div`
    ${props =>
        props.minified
            ? `background-image: linear-gradient(to bottom, ${props.theme.walletBackground}, ${props.theme.walletInfoContainer})`
            : `background: ${props.theme.walletInfoContainer}`};
    padding: 12px 20px;
    padding-top: 0px;
    box-sizing: border-box;
    transition: all 0.5s ease-in-out;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;
