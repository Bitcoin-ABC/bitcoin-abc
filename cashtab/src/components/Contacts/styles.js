// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const ContactList = styled.div`
    margin-top: 24px;
    padding: 12px;
    display: flex;
    flex-direction: column;
    width: 100%;
    align-items: center;
    gap: 12px;
    color: ${props => props.theme.contrast};
    box-sizing: border-box;
    *,
    *:before,
    *:after {
        box-sizing: inherit;
    }
`;

export const ContactsPanel = styled.div`
    display: flex;
    flex-direction: column;
    padding: 12px;
    width: 100%;
    background-color: ${props => props.theme.panel};
    border-radius: 9px;
    margin-bottom: 12px;
`;

export const Row = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    justify-content: space-between;
    gap: 12px;
    flex-wrap: wrap;
    border-bottom: 0.5px solid ${props => props.theme.separator};
    padding: 6px 0;
`;
export const ButtonRow = styled.div`
    display: flex;
    width: 100%;
    align-items: center;
    gap: 12px;
`;

export const ContactListName = styled.div`
    display: flex;
    text-align: left;
    word-break: break-word;
`;

export const ButtonPanel = styled.div`
    display: flex;
    gap: 9px;
    align-items: baseline;
`;
