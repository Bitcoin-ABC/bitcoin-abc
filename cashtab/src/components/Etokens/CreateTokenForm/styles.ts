// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const Form = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
`;
export const SwitchRow = styled.div`
    display: flex;
    flex-direction: row;
    gap: 12px;
    align-items: center;
`;
export const SwitchLabel = styled.div`
    text-align: left;
    color: ${props => props.theme.contrast};
    font-size: 18px;
`;
export const EditIcon = styled.div`
    cursor: pointer;
    color: ${props => props.theme.contrast};
    &:hover {
        color: ${props => props.theme.eCashBlue};
    }
    word-wrap: break-word;
`;
export const TokenCreatedLink = styled.a`
    color: ${props => props.theme.walletBackground};
    text-decoration: none;
`;
export const IconModalForm = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
    justify-content: center;
`;
export const IconModalRow = styled.div`
    display: flex;
    width: 100%;
    gap: 3px;
`;
export const SliderLabel = styled.div`
    color: ${props => props.theme.contrast};
`;
export const SliderBox = styled.div`
    width: 100%;
`;
export const CropperContainer = styled.div`
    height: 200px;
    position: relative;
`;
export const CreateTokenTitle = styled.h3`
    color: ${props => props.theme.contrast};
`;
export const TokenCreationSummaryTable = styled.div`
    color: ${props => props.theme.contrast};
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    gap: 3px;
`;
export const SummaryRow = styled.div`
    display: flex;
    gap: 12px;
    justify-content: flex-start;
    align-items: start;
    width: 100%;
`;
export const TokenParam = styled.div`
    word-break: break-word;
`;
export const ButtonDisabledMsg = styled.div`
    font-size: 14px;
    color: ${props => props.theme.forms.error};
    word-break: break-all;
`;
