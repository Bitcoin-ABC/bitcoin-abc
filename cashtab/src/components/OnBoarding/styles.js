// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const WelcomeCtn = styled.div`
    margin-top: 20px;
    padding: 0px 30px;
    color: ${props => props.theme.contrast};
    h2 {
        color: ${props => props.theme.contrast};
    }
`;

export const WelcomeText = styled.p`
    width: 100%;
    font-size: 16px;
    margin-bottom: 60px;
    text-align: left;
`;

export const WelcomeLink = styled.a`
    text-decoration: underline;
    color: ${props => props.theme.eCashBlue};
    :hover {
        color: ${props => props.theme.eCashPurple} !important;
        text-decoration: underline !important;
    }
`;
