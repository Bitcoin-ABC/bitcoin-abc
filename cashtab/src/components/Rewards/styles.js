// Copyright (c) 2024 The Bitcoin developers
// Distributed under the MIT software license, see the accompanying
// file COPYING or http://www.opensource.org/licenses/mit-license.php.

import styled from 'styled-components';

export const Wrapper = styled.div`
    width: 100%;
    margin-top: 24px;
`;
