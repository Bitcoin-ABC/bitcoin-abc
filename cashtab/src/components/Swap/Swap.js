import React from 'react';
import styled from 'styled-components';
import { WalletContext } from 'utils/context';
import { getWalletState } from 'utils/cashMethods';
import { SmartButton } from 'components/Common/PrimaryButton';
import {
    WalletInfoCtn,
    ZeroBalanceHeader,
    SidePaddingCtn,
} from 'components/Common/Atoms';
import BalanceHeader from 'components/Common/BalanceHeader';
import BalanceHeaderFiat from 'components/Common/BalanceHeaderFiat';
import WalletLabel from 'components/Common/WalletLabel';
import { CustomCollapseCtn } from 'components/Common/StyledCollapse';
import { isValidSideshiftObj } from 'utils/validation';
import { AlertMsg } from 'components/Common/Atoms';
import appConfig from 'config/app';

export const SwapCtn = styled.div`
    width: 100%;
    h2 {
        color: ${props => props.theme.contrast};
        margin: 0 0 20px;
        margin-top: 10px;
    }
`;

const Swap = () => {
    const ContextValue = React.useContext(WalletContext);
    const { wallet, cashtabSettings, changeCashtabSettings, fiatPrice } =
        ContextValue;
    const walletState = getWalletState(wallet);
    const { balances } = walletState;
    const sideshift = window.sideshift;

    return (
        <>
            <SwapCtn>
                <WalletInfoCtn>
                    <WalletLabel
                        name={wallet.name}
                        cashtabSettings={cashtabSettings}
                        changeCashtabSettings={changeCashtabSettings}
                    ></WalletLabel>
                    {!balances.totalBalance ? (
                        <ZeroBalanceHeader>
                            You currently have 0 {appConfig.ticker}
                            <br />
                            Deposit some funds to use this feature
                        </ZeroBalanceHeader>
                    ) : (
                        <>
                            <BalanceHeader
                                balance={balances.totalBalance}
                                ticker={appConfig.ticker}
                                cashtabSettings={cashtabSettings}
                            />

                            <BalanceHeaderFiat
                                balance={balances.totalBalance}
                                settings={cashtabSettings}
                                fiatPrice={fiatPrice}
                            />
                        </>
                    )}
                </WalletInfoCtn>
            </SwapCtn>
            <br />
            <SidePaddingCtn>
                <CustomCollapseCtn panelHeader="SideShift">
                    {isValidSideshiftObj(sideshift) ? (
                        <>
                            <br />
                            <SmartButton onClick={() => sideshift.show()}>
                                Open SideShift
                            </SmartButton>
                        </>
                    ) : (
                        <AlertMsg>`Error: Unable to load SideShift`</AlertMsg>
                    )}
                </CustomCollapseCtn>
            </SidePaddingCtn>
        </>
    );
};

export default Swap;
