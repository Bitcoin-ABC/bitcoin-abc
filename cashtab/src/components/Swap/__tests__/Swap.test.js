import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import Swap from 'components/Swap/Swap';
import {
    walletWithBalancesAndTokens,
    walletWithBalancesMock,
    walletWithoutBalancesMock,
    walletWithBalancesAndTokensWithCorrectState,
} from '../../Home/__mocks__/walletAndBalancesMock';
import { BrowserRouter as Router } from 'react-router-dom';
import { WalletContext } from 'utils/context';

beforeEach(() => {
    // Mock method not implemented in JSDOM
    // See reference at https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation(query => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(), // Deprecated
            removeListener: jest.fn(), // Deprecated
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
});

test('Wallet without XEC balance', () => {
    const component = renderer.create(
        <WalletContext.Provider value={walletWithoutBalancesMock}>
            <ThemeProvider theme={theme}>
                <Router>
                    <Swap />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Wallet with XEC balances', () => {
    const component = renderer.create(
        <WalletContext.Provider value={walletWithBalancesMock}>
            <ThemeProvider theme={theme}>
                <Router>
                    <Swap />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Wallet with XEC balances and tokens', () => {
    const component = renderer.create(
        <WalletContext.Provider value={walletWithBalancesAndTokens}>
            <ThemeProvider theme={theme}>
                <Router>
                    <Swap />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Wallet with XEC balances and tokens and state field', () => {
    const component = renderer.create(
        <WalletContext.Provider
            value={walletWithBalancesAndTokensWithCorrectState}
        >
            <ThemeProvider theme={theme}>
                <Router>
                    <Swap />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('Without wallet defined', () => {
    const withoutWalletDefinedMock = {
        wallet: {},
        balances: { totalBalance: 0 },
        loading: false,
    };
    const component = renderer.create(
        <WalletContext.Provider value={withoutWalletDefinedMock}>
            <ThemeProvider theme={theme}>
                <Router>
                    <Swap />
                </Router>
            </ThemeProvider>
        </WalletContext.Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
