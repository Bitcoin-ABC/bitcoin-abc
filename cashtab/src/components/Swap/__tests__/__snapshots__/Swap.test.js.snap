// Jest Snapshot v1, https://goo.gl/fbAQLP @generated

exports[`Wallet with XEC balances 1`] = `
Array [
  <div
    className="sc-gisBJw lhAjrX"
  >
    <div
      className="sc-jWBwVP cKvBni"
    >
      <div
        className="sc-jhAzac kaAIpD"
      >
        <h4
          className="sc-fBuWsC eRpPSf"
        >
          MigrationTestAlpha
        </h4>
        <a
          href="/configure"
          onClick={[Function]}
        >
          <svg
            className="sc-jTzLTM bTdWCF"
          >
            edit.svg
          </svg>
        </a>
        <div>
           
          <button
            aria-checked={false}
            className="ant-switch ant-switch-small"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <svg
                className="sc-kAzzGY fUTkYj"
              >
                eye-invisible.svg
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="sc-jAaTju Vyslf"
      >
        You currently have 0 
        XEC
        <br />
        Deposit some funds to use this feature
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="sc-eHgmQL kvZwAq"
  >
    <div
      className="sc-eqIVtm jMsian"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-caSCKo xvorO"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-dVhcbM fghLuF"
              >
                SideShift
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Wallet with XEC balances and tokens 1`] = `
Array [
  <div
    className="sc-gisBJw lhAjrX"
  >
    <div
      className="sc-jWBwVP cKvBni"
    >
      <div
        className="sc-jhAzac kaAIpD"
      >
        <h4
          className="sc-fBuWsC eRpPSf"
        >
          MigrationTestAlpha
        </h4>
        <a
          href="/configure"
          onClick={[Function]}
        >
          <svg
            className="sc-jTzLTM bTdWCF"
          >
            edit.svg
          </svg>
        </a>
        <div>
           
          <button
            aria-checked={false}
            className="ant-switch ant-switch-small"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <svg
                className="sc-kAzzGY fUTkYj"
              >
                eye-invisible.svg
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="sc-jAaTju Vyslf"
      >
        You currently have 0 
        XEC
        <br />
        Deposit some funds to use this feature
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="sc-eHgmQL kvZwAq"
  >
    <div
      className="sc-eqIVtm jMsian"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-caSCKo xvorO"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-dVhcbM fghLuF"
              >
                SideShift
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Wallet with XEC balances and tokens and state field 1`] = `
Array [
  <div
    className="sc-gisBJw lhAjrX"
  >
    <div
      className="sc-jWBwVP cKvBni"
    >
      <div
        className="sc-jhAzac kaAIpD"
      >
        <h4
          className="sc-fBuWsC eRpPSf"
        >
          MigrationTestAlpha
        </h4>
        <a
          href="/configure"
          onClick={[Function]}
        >
          <svg
            className="sc-jTzLTM bTdWCF"
          >
            edit.svg
          </svg>
        </a>
        <div>
           
          <button
            aria-checked={false}
            className="ant-switch ant-switch-small"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <svg
                className="sc-kAzzGY fUTkYj"
              >
                eye-invisible.svg
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="sc-cMljjf coNPHF"
      >
        <span
          className="sc-Rmtcm iyxXlA"
        >
           
          0.06
           
          XEC
           
        </span>
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="sc-eHgmQL kvZwAq"
  >
    <div
      className="sc-eqIVtm jMsian"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-caSCKo xvorO"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-dVhcbM fghLuF"
              >
                SideShift
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Wallet without XEC balance 1`] = `
Array [
  <div
    className="sc-gisBJw lhAjrX"
  >
    <div
      className="sc-jWBwVP cKvBni"
    >
      <div
        className="sc-jhAzac kaAIpD"
      >
        <h4
          className="sc-fBuWsC eRpPSf"
        >
          MigrationTestAlpha
        </h4>
        <a
          href="/configure"
          onClick={[Function]}
        >
          <svg
            className="sc-jTzLTM bTdWCF"
          >
            edit.svg
          </svg>
        </a>
        <div>
           
          <button
            aria-checked={false}
            className="ant-switch ant-switch-small"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <svg
                className="sc-kAzzGY fUTkYj"
              >
                eye-invisible.svg
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="sc-jAaTju Vyslf"
      >
        You currently have 0 
        XEC
        <br />
        Deposit some funds to use this feature
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="sc-eHgmQL kvZwAq"
  >
    <div
      className="sc-eqIVtm jMsian"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-caSCKo xvorO"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-dVhcbM fghLuF"
              >
                SideShift
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`Without wallet defined 1`] = `
Array [
  <div
    className="sc-gisBJw lhAjrX"
  >
    <div
      className="sc-jWBwVP cKvBni"
    >
      <div
        className="sc-jhAzac kaAIpD"
      >
        <a
          href="/configure"
          onClick={[Function]}
        >
          <svg
            className="sc-jTzLTM bTdWCF"
          >
            edit.svg
          </svg>
        </a>
        <div>
           
          <button
            aria-checked={false}
            className="ant-switch ant-switch-small"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="switch"
            type="button"
          >
            <div
              className="ant-switch-handle"
            />
            <span
              className="ant-switch-inner"
            >
              <svg
                className="sc-kAzzGY fUTkYj"
              >
                eye-invisible.svg
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        className="sc-jAaTju Vyslf"
      >
        You currently have 0 
        XEC
        <br />
        Deposit some funds to use this feature
      </div>
    </div>
  </div>,
  <br />,
  <div
    className="sc-eHgmQL kvZwAq"
  >
    <div
      className="sc-eqIVtm jMsian"
    >
      <div
        className="ant-collapse ant-collapse-icon-position-start sc-caSCKo xvorO"
        role={null}
        style={
          Object {
            "marginBottom": "24px",
          }
        }
      >
        <div
          className="ant-collapse-item"
        >
          <div
            aria-disabled={false}
            aria-expanded={false}
            className="ant-collapse-header"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="ant-collapse-expand-icon"
              onClick={null}
            >
              <span
                aria-label="right"
                className="anticon anticon-right ant-collapse-arrow"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <span
              className="ant-collapse-header-text"
              onClick={null}
            >
              <div
                className="sc-dVhcbM fghLuF"
              >
                SideShift
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
